spec/controllers/assignments_controller_spec.rb

```diff
@@ -99,20 +99,15 @@
 
   context 'CSV_Downloads' do
     let(:csv_options) do
-      {
-        type: 'text/csv',
-        filename: 'assignment_list.csv',
-        disposition: 'attachment'
-      }
+      { type: 'text/csv', filename: "assignments_list_#{Time.now.strftime('%Y%m%d')}.csv" }
     end
-
-    before :each do
+    let(:assignment) {create(:assignment)}
+    #before :each do
       # for some reason, assignments aren't always cleared from the db
       # before these tests
-      Assignment.all.each do |asn|
-        asn.delete
-      end
-      @assignment = FactoryBot.create(:assignment)
+    #  Assignment.all.each do |asn|
+    #    asn.delete
+    #  end
     end
```

Comment:
You need to comment this out as well, and uncomment the end at the bottom of this file. (I think this is the cause of the second error, with `let(:assignment)` ending its scope here.